### ISSUE TITLE
Bold fonts in not working correctly in UI?

### ISSUE THREAD

**Issue by GoogleCodeExporter** [rakibtg/gdipp#25]
```
What steps will reproduce the problem?
1. Configure Windows XP to show bold fonts in window UI elements


What is the expected output? What do you see instead?
Bold fonts should be used, however normal fonts, with increased character
spacing are shown instead.  Please refer to the first two attachments
showing the Firefox titlebar, with/without bold fonts.

What version of gdipp are you using? On what operating system (with
bitness)?

gdipp 0.7.2
Windows XP SP3 32bit (Ja)


Please provide any additional information below.

First, thank you for fixing the font manager to work correctly in Windows
XP, and for cleaning-up the online documentation.  Everything appears to be
functioning correctly, however elements of the Windows GUI that are set to
be shown in Bold display as normal text, yet with larger spacing between
characters.

In order to set bold fonts, use the Windows XP 'Appearance' dialogue (from
the right-mouse 'Desktop Properties' applet), select a UI element, then
enable/disable the bold control ('B' button).  Note that I am using 'Lucida
Grande' font, however the same thing happens with standard Windows fonts
such as Tahoma etc.

Cheers,
James
x
```

Original issue reported on code.google.com by `jmbat...@gmail.com` on 29 Mar 2010 at 2:11

Attachments:
- [gdipp.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-0/gdipp.png)
- [gdi++.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-0/gdi++.png)
- [gdipp2.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-0/gdipp2.png)

**Comment by GoogleCodeExporter** [comment]
```
Me too.

I using Windows 7 x64. some fonts didn't show bold texts in Internet 
Explorer(or all 
of program). Character Spacing is increased but not bold texts.

Here, I attach screenshots.
gdipp_bold_1.png, increased Character Spacing but we can't look bold texts.
gdipp_bold_2.png is not rendered screenshot. We can look bold texts.
gdipp_bold_3.png is rendered by gdipp. Where are bold texts?

Legacy CJK from Windows default truetype fonts didn't make bold font.

MS Gothic, MS PGothic, MS UI Gothic
MS Mincho, MS PMincho
Gulim, Gulimche
Dotum, Dotumche
Batang, Batangche
Gungsuh, Gungsuhche
MingLiu, PMingLiu
SimSun, NSimSun
SimHei
KaiTi

and so on.

```

Original comment by `basic2...@gmail.com` on 1 Apr 2010 at 2:43

Attachments:
- [gdipp_bold_1.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-1/gdipp_bold_1.png)
- [gdipp_bold_2.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-1/gdipp_bold_2.png)
- [gdipp_bold_3.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-1/gdipp_bold_3.png)

**Comment by GoogleCodeExporter** [comment]
```
This problem occurs only if there is no separate Bold style file for the font 
family.
```

Original comment by `crendk...@gmail.com` on 7 Apr 2010 at 5:29
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Ah, I see.

Is this something that I can fix myself (by adjusting the font file), or do you 
intend 
to add support for such fonts in future releases?

Many thanks,
James
x
```

Original comment by `jmbat...@gmail.com` on 7 Apr 2010 at 5:39

**Comment by GoogleCodeExporter** [comment]
```
I have identified the problem. I will fix it in the next release.
```

Original comment by `crendk...@gmail.com` on 7 Apr 2010 at 10:20

**Comment by GoogleCodeExporter** [comment]
```
Great stuff - many thank!

James
x
```

Original comment by `jmbat...@gmail.com` on 7 Apr 2010 at 12:37

**Comment by GoogleCodeExporter** [comment]
```
I wonder if the fix made it too bold? It was quite unreadable on 0.7.4 beta so 
I 
reverted to 0.7.3
```

Original comment by `syoc...@gmail.com` on 11 Apr 2010 at 11:50

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Well, the UI elements that were not bold previous are now rendered as bold - I 
have 
made some comparisons with gdi++ the bold weight appears to match.

However, for some reason the normal (non-bold) style is now rendered very 
faintly, 
yet only in some programmes.  The UI in Thunderbird looks great, however 
Chromium and 
Explorer appear very soft.

Again, I have attached screenshots (gdipp 0.7.4, gdipp 0.7.3, and gdi++) for 
reference.

Cheers,
James
x
```

Original comment by `jmbat...@gmail.com` on 12 Apr 2010 at 12:03

Attachments:
- [gdipp - 0.7.4 - Explorer.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-8/gdipp - 0.7.4 - Explorer.png)
- [gdipp - 0.7.3 - Explorer.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-8/gdipp - 0.7.3 - Explorer.png)
- [gdi++ - Explorer.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-8/gdi++ - Explorer.png)
- [gdipp - 0.7.4 - Chromium options.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-8/gdipp - 0.7.4 - Chromium options.png)
- [gdipp - 0.7.3 - Chromium options.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-8/gdipp - 0.7.3 - Chromium options.png)
- [gdi++ - Chromium options.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-8/gdi++ - Chromium options.png)

**Comment by GoogleCodeExporter** [comment]
```
Regarding to "too bold" problem:

More comprehensive answers can be found here:
http://code.google.com/p/gdipp/issues/detail?id=34.

"Since 0.7.4, gdipp has the ability to render ANY weight the program requests." 
If
your programs request that, gdipp obeys.

Regarding to "too soft" question:

0.7.4 reduces the default shadow alpha. You can try to increase it.
```

Original comment by `crendk...@gmail.com` on 12 Apr 2010 at 4:23

**Comment by GoogleCodeExporter** [comment]
```
Thank you, and yes I just read about the flexible font weighting ability.
Do you know if gdi++ offers such functionality?

Would it be possible to request an 'old style' option to restrict rendering to 
the 
simple 400/700 system used by ClearType?

>0.7.4 reduces the default shadow alpha. You can try to increase it.

Yes, I noticed this also, however I always set alpha to 0 immediately after 
reinstalling gdipp (I personally don't find the shadowing improves legibility), 
so I 
don't believe this is the issue.  It would appear that the normal weight on 
some UI 
elements is simply too 'light' to begin with - I suspect this may also relate 
to the 
flexible weighting functionality outlined above.

Cheers,
James
x
```

Original comment by `jmbat...@gmail.com` on 12 Apr 2010 at 4:31

**Comment by GoogleCodeExporter** [comment]
```
If I can see enough evidence showing that GDI really ignores other weight 
values, I
will revert to emulate GDI's behavior. This is only a regression of the current
weight ability, thus can be done easily. Still, compatibility is one of the most
important thing gdipp considers. I just wonder why GDI exhibits such ignoring
behavior. Maybe Microsoft's burden of compatibility is much more heavier than 
what we
can imagine.

If program requests weight as 200 or 100, gdipp will render it very faintly.
```

Original comment by `crendk...@gmail.com` on 12 Apr 2010 at 4:54

**Comment by GoogleCodeExporter** [comment]
```
I see.

Do you know what weight the Windows explorer icon text (including desktop), or 
perhaps the Chromium options window in the screenshots above are  'requested' 
to be 
rendered at?

Cheers,
James
x
```

Original comment by `jmbat...@gmail.com` on 12 Apr 2010 at 5:02

**Comment by GoogleCodeExporter** [comment]
```
400 for Explorer. I think Chrome should be the same. gdipp works perfect in my 
own
machine.
```

Original comment by `crendk...@gmail.com` on 12 Apr 2010 at 7:42

**Comment by GoogleCodeExporter** [comment]
```
Should be fixed in 0.7.5.
```

Original comment by `crendk...@gmail.com` on 16 Apr 2010 at 7:14
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Thank you.  0.7.5 fixes the issue with some UI elements being rendered too 
softly.

However, I note that my window titles are now too bold.
I believe gdipp may be rendering 'faux-bold' (i.e. fonts that do not have a 
separate 
bold file) too heavily.

The OS X theme I am using utilises the 'Lucida Grande' font, yet strangely, 
despite 
the separate 'Lucida Grande Bold' file being available, the system still 
chooses to 
use a faux-bold version of the normal 'Lucida Grande' font.  I wonder if I need 
to 
edit the .msstyle theme or even the .ttf file in some way to specify that the 
bold 
version of the font is part of the same family?

Cheers,
James
x
```

Original comment by `jmbat...@gmail.com` on 19 Apr 2010 at 12:05

Attachments:
- [gdipp - 0.7.5 - Explorer.png](https://storage.googleapis.com/google-code-attachments/gdipp/issue-25/comment-15/gdipp - 0.7.5 - Explorer.png)
